DEVELOPER.md

```diff
@@ -226,31 +238,33 @@ on [ci-firmwares](https://github.com/iot-lab/ci-firmwares)
 Testing your implementation
 ===========================
 
-We've provided Dockerfiles with all the dependencies and a Makefile
-for you to build and use them. (if using MacOS, see Appendix below)
+For unit-testing and integration-testin it's recommended you use the 
+provided Docker images, but you can test without Docker if you want.
+
+### Unit tests
+
+The purpose of unit tests is to verify your Python code,
+independently from connections to a physical board.
+
+### Integration tests
+
+The purpose of integration tests is to verify that your flashing tool
+and firmwares behave as expected when called by the gateway code. 
 
 Using Docker
 ------------
 
-If you want to run tests with the Docker images, first build them:
-
-    make build-docker-image-test
+You should read [DOCKER.md](DOCKER.md) for prerequisites
 
-### Unit tests
+Once udev rules and the base `iot-lab-gateway` image is built: build the testing image:
```

Comment:
This sentence is weird.

Note that there's a target to setup the udev-rules on the host computer:

```
make setup-udev-rules
```

So maybe say something like:

```md
Setup udev-rules
----------------

The udev-rules must be setup on the host computer in order to map the custom iotlab TTY in the docker container:

    make setup-udev-rules

Using Docker
------------

You should read [DOCKER.md](DOCKER.md) for prerequisites

Build the testing image:

    make build-docker-image-test

```

Reply:
ok, I've corrected that part